R/formatter.r

```diff
@@ -1,3 +1,72 @@
+#' Number formatter: a generic formatter for numbers
+#'
+#' @return \code{number_format} returns a function with single parameter
+#' \code{x}, a numeric vector, that returns a character vector
+#' @param x a numeric vector to format
+#' @param accuracy number to round to, \code{NULL} for automatic guess
+#' @param scale a scaling factor (useful if the underlying is on another
+#' scale, e.g. for computing percentages or thousands)
+#' @param prefix,suffix symbols to display before and after value
+#' @param big.mark character used between every 3 digits to separate thousands
+#' @param decimal.mark the character to be used to indicate the numeric
+#' decimal point
+#' @param trim logical, if \code{FALSE}, values are right-justified to a common
+#' width (see \code{\link[base]{format}})
+#' @param ... other arguments passed on to \code{\link[base]{format}}
+#' @export
+#' @examples
+#' v <- c(12.3, 4, 12345.789, 0.0002)
+#' number(v)
+#' number(v, big.mark = ",")
+#' number(v, accuracy = .001)
+#' number(v, accuracy = .001, decimal.mark = ",")
+#' number(v, accuracy = .5)
+#'
+#' my_format <- number_format(big.mark = "'", decimal.mark = ",")
+#' my_format(v)
+#'
+#' # Per mille
+#' per_mille <- number_format(scale = 1000,
+#'                            suffix = "\u2030",
+#'                            accuracy = .1)
+#' per_mille(v)
+number_format <- function(accuracy = 1, scale = 1, prefix = "",
+                          suffix = "", big.mark = " ", decimal.mark = ".",
+                          trim = TRUE, ...) {
+  function(x) number(
+      x, accuracy, scale, prefix,
```

Comment:
Can you please name arguments here (apart from `x`). That makes it safer if we later change the arguments to `number()`.

Reply:
corrected